### PR TITLE
Add type safe opSync and opAsync

### PR DESCRIPTION
**Pull request by espindola**
I have fixed most type errors already. This fixes a null X undefined
error and adds a FIXME for the last one I know of.

### CODE REVIEW

File: api/src/chisel.ts [hunk 1]
```diff
@@ -481,19 +489,19 @@ export class ChiselCursor<T> {
...
                     while (true) {
-                        const properties = await Deno.core.opAsync(
+                        const properties = await opAsync(
                             "op_chisel_query_next",
                             rid,
-                        );
+                        ) as T | null; // FIXME: This is wrong, we can get less than T with ColumnsSelect.
 
-                        if (properties == undefined) {
+                        if (properties === null) {
```
Comment (BearLemma) [inline api/src/chisel.ts:504]: Why is is this changed to null?

**Comment by dekimir** [comment]
So `(a?: unknown, b?: unknown): unknown` is type safe? ;)

**Comment by espindola** [comment]
It is. You have to explicitly convert unknown to another type before
using, so we can just search for type asserts to see where errors might
be.

The existing Deno.core.opSync/opAsync use any instead of unknown.

Cheers,
Rafael

**Comment by BearLemma** [comment]
Interesting, I wrote the exact same thing via email, but it didn't work for me for some reason 🤔  @espindola you replied via email, right? And did you click reply all or just reply?

File: api/src/chisel.ts (hunk 1 above)
Comment (espindola) [reply api/src/chisel.ts:504]: Because it is more precise. We were returning null already, but the comparison would pass because of the == instead of ===

**Comment by BearLemma** [comment]
I think I did that as well :thinking:. Let's see if it works this time.

**Review by BearLemma — APPROVED** [review]

**Comment by espindola** [comment]
Because that is what we get from rust. The old code worked because it
was using == instead of ===.

Cheers,
Rafael

**Pull request merged**